### DISCUSSION TITLE
posts/hamsterkombat-clicker/

### DISCUSSION THREAD

**Discussion by giscus** [serezhka/serezhka.github.io#4] [General]
# posts/hamsterkombat-clicker/

Intro

https://serezhka.github.io/posts/hamsterkombat-clicker/

**Comment by portg80** [comment]
The script pauses when I go to a new tab, I wanted to fix this by running a typermonkey script, but that doesn't work because you need to select javascript context "clicker (hamsterkombat.io)" and I don't know how to specify that through the script, how can I go about it?

**Comment by serezhka** [comment]
Hi @portg80, I decided to keep the script as simple as possible. Because taps in total bring quite a few coins. Good luck 🪙🍀

**Comment by repon06** [comment]
clicker (hamsterkombat.io)
VM1045:1 Uncaught ReferenceError: clicker is not defined
    at <anonymous>:1:1

**Reply by repon06** [reply]
sorry, all so good)

**Comment by Teraskull** [comment]
You can also just edit the `__telegram__initParams` in the Session Storage like this:

![Screenshot 2024-06-05 091810](https://github.com/community/community/assets/24798198/20529d9d-565d-4749-b17c-e41be37654d9)

Edit `tgWebAppPlatform` to be `ios` and then click "Reload Page" from the 3 dot menu.

The only downside is that `tgWebAppPlatform` gets overwritten on page refresh, but I believe that also can be fixed by updating it with a userscript.

Another way is to copy the iframe `src` URL in the image below, while the game is running (will be a QR code on black background) and paste it in a new tab in the same browser where you're logged into Telegram.

![Screenshot 2024-06-05 093523](https://github.com/community/community/assets/24798198/1949b797-d8cf-4aef-bdc5-934dc58ad69e)

In that URL change `tgWebAppPlatform=web` (It can also be `weba`) to `tgWebAppPlatform=ios` and open it. Now the authorization is saved between refreshes.

Also, all allowed values for `tgWebAppPlatform` are `android`, `android_x`, and `ios`.

The authorization token in that URL will get invalidated eventually, either after some time (I don't know exactly when, didn't happen to me yet), or after you open the game inside Telegram. Then you have to copy and edit the URL again to get the refreshed authorization token.

Still better than running an Android emulator though.

Thanks for your info, it was helpful!

**Reply by Teraskull** [reply]
Also, when you're running the web app from the `hamstercombat.io` domain directly, you can just do this in the console instead of an autoclicker:

```js
localStorage.setItem("hk_sync_taps", 10000);
```

Where `10000` is your energy limit or higher.
It will mark the energy as tapped and ready to sync, so just refresh the page and the energy level will go back to 0.

This won't work from Telegram Web because of CORS though.

I did write a userscript to allow running the web app from Telegram Web, and add an "Open  in New Tab" button to the 3 dot menu:

https://greasyfork.org/en/scripts/497148-hamster-kombat-desktop

No Javascript overriding is needed this way, only a script manager, like Tampermonkey or Violentmonkey.

**Reply by MichaelNguyen1904** [reply]
Thank for your instruction.

**Comment by Armagedon622** [comment]
Getting "Request error, unable to sync" as of today.

**Comment by Teraskull** [comment]
@Armagedon622 That's just Hamster Kombat acting up, nothing to do with this modification.

**Reply by Armagedon622** [reply]
Well its the second day now that its acting up :/

**Reply by Armagedon622** [reply]
And sorry that i didnt clarify, the mod itself is not the cause, but maybe the resource override is the problem?Ive been using it for a while now, so maybe the link is outdated for it or something

**Reply by Teraskull** [reply]
Nope, the `hamstercombat.io` server is currently down, which is required for the web app authorization.

The resource override does not break anything and is not outdated. All the script does is replaces the platform string from `web` to `ios`.

**Comment by Armagedon622** [comment]
Got it

**Comment by HEROMASTANI** [comment]
Hi , I did all, but ,Override content is not available , I installed 'Resource Override"  and done as said:

**Comment by meti-0763** [comment]
VM3459:5 Uncaught 
TypeError: Cannot read properties of undefined (reading 'getElementsByTagName')
    at fn (<anonymous>:5:86)
    at <anonymous>:11:7
    at <anonymous>:12:3
fn	@	VM3459:5
(anonymous)	@	VM3459:11
(anonymous)	@	VM3459:12 


error :(

**Reply by mohsen202** [reply]
i have the same error. please help me. thanks

**Reply by serezhka** [reply]
Please, make sure that you chose the correct javascript context. It should be 'clicker'

**Reply by NinhFPT14** [reply]
I encountered this situation, by chance I checked the element and ran the javascrip command again in the console and it worked again

**Comment by NinhFPT14** [comment]
I have found this extension to be faster https://www.google.com/search?q=ch%E1%BA%A1y+hamster+kombat+chorme&oq=ch%E1%BA%A1y+&gs_lcrp=EgZjaHJvbWUqBggAEEUYOzIGCAAQRRg7MggIARBFGCcYOzIGCAIQRRg7MgYIAxBFGEAyBggEEEUYOTIGCAUQIxgnMgwIBhAAGEMYgAQYigUyDAgHEAAYQxiABBiKBdIBCDE3OTZqMGo3qAIAsAIA&sourceid=chrome&ie=UTF-8#fpstate=ive&vld=cid:f45747b6,vid:29BjCQMFNww,st:0

**Comment by bobbrysonn** [comment]
Genius! Works like charm

**Comment by serezhka** [comment]
The new url of telegram-web-app is: https://hamsterkombat.io/js/telegram-web-app.js?v=7.6
You need just to update it in your overrides or create a new rule 
<img width="791" alt="Screenshot 2024-07-03 at 23 19 46" src="https://github.com/serezhka/serezhka.github.io/assets/476359/8ee7eeee-66b7-4683-ab38-d209fc53c403">

**Comment by Blackjaguareats** [comment]
Bot the web telegram can't open . It make a screenshot with play on your mobile

**Comment by AKTanara** [comment]
new URL:
https://hamsterkombatgame.io/js/telegram-web-app.js?v=7.6

**Comment by tramvayabakan** [comment]
they added new mini-game. it keeps loading but doesnt open. can you update it please?

**Comment by khode-mohsen** [comment]
for those how cant open in telegram web , use this url in first one ...

https://hamsterkombatgame.io/js/telegram-web-app.js?v=7.6

**Comment by MosMosi12345-cloud** [comment]
Could you please add this tutorial to the heart of the film? I would really appreciate it because I don't know foreign languages. I kindly request, thank you very much if you could also provide a tutorial on AutoClick for hamsters.

**Comment by MosMosi12345-cloud** [comment]
Could you please add this tutorial to the heart of the film? I would really appreciate it because I don't know foreign languages. I kindly request, thank you very much if you could also provide a tutorial on AutoClick for hamsters.
